jina/types/document/generators.py

```diff
@@ -31,9 +31,9 @@ def from_ndarray(
 
     if shuffle:
         # shuffle for random query
-        array = np.take(array, np.random.permutation(array.shape[0]), axis=axis)
+        array = np.take(array, np.random.permutation(array.shape[axis]), axis=axis)
```

Comment:
maybe it's even better to:
```suggestion
        array = np.random.shuffle(array) 
```

and parameterize the unit test

Reply:
and remove the `axis`, i don't see why it's needed